hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2175,7 +2175,8 @@ public boolean getPushGatewayRandomJobNameSuffix() {
   }
 
   public String getMetricReporterMetricsNamePrefix() {
-    return getStringOrDefault(HoodieMetricsConfig.METRICS_REPORTER_PREFIX);
+    // Metrics prefixes should not have a dot as this is usually a separator
+    return getStringOrDefault(HoodieMetricsConfig.METRICS_REPORTER_PREFIX).replaceAll("\\.", "_");
```

Comment:
Can we just report the invalid format and throws exception ?

Reply:
I see why we do this. If the metrics prefix has "." (dots), it might cause issues w/ dashbaords. So, may be not a bad idea. but it might break someone who was using it so far (by mistake)
@rmahindra123 @harsh1231 : do you folks have any thoughts here.

Reply:
@rmahindra123 @harsh1231 Do you have any thoughts on this change?

Reply:
I think it should be ok as long as well call out the change. Most metrics systems are append-only, so there will be two metrics for the same purpose for some time. Downstream consumers will have to make changes in the query to pick up the newer format.